sdks/java/io/thrift/src/main/java/org/apache/beam/sdk/io/thrift/ThriftSchema.java

```diff
@@ -90,17 +95,17 @@
  *       parameter exists.
  *   <li>All non-union types have a corresponding java field with the same name for every field in
  *       the original thrift source file.
- *   <li>The underlying {@link FieldMetaData#getStructMetaDataMap(Class) metadata maps} are {@link
- *       java.util.EnumMap enum maps}, so the natural order of the field keys is preserved.
```

Comment:
This is still the case with thrift metadata, but it's no longer a "strong assumption" for this class, as now the field order is infered from the schema, allowing for field reordering.